tcp_endpoint/src/tcp_endpoint/RosUnityHandshakeService.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+
+import rospy
+from tcp_endpoint.srv import RosUnityHandshake, RosUnityHandshakeResponse
+
+class RosUnityHandshakeService:
+    """
+    Class to auto-detect Unity IP.
+    """
+    def __init__(self, tcp_sender):
+        """
+        Args:
+            tcp_sender:     sends messages to Unity
+        """
+        self.srv_class = RosUnityHandshake._request_class()
+        self.tcp_sender = tcp_sender
+        self.incoming_ip = ''
+
+    # The client thread lets us know what the incoming IP is, so we can use it later
+    def set_thread(self, tcp_client_thread):
```

Comment:
This function name could be more descriptive - looks closer to `bind_to_client_ip` or similar?  It's certainly not setting a variable named thread.

Reply:
FYI, main branch is protected so I literally cannot push an unapproved PR.

Reply:
Ok gotcha - that is unfortunate!